src/snowflake/cli/_plugins/connection/commands.py

```diff
@@ -355,36 +354,23 @@ def set_default(
     return MessageResult(f"Default connection set to: {name}")
 
 
-@app.command(requires_connection=False)
+@app.command(requires_connection=True)
 def generate_jwt(
-    account: str = typer.Option(
-        None,
-        "--account",
-        "-a",
-        "--accountname",
-        help="Account name to use when authenticating with Snowflake.",
-        show_default=False,
-    ),
-    user: str = typer.Option(
-        None,
-        "--user",
-        "-u",
-        "--username",
-        show_default=False,
-        help="Username to connect to Snowflake.",
-    ),
-    private_key_file: Path = typer.Option(
-        None,
-        "--private-key",
-        "--private-key-path",
-        "-k",
-        help="Path to file containing private key",
-        dir_okay=False,
-        exists=True,
-    ),
     **options,
 ) -> CommandResult:
     """Generate a JWT token, which will be printed out and displayed.."""
+    connection_details = get_cli_context().connection_context.update_from_config()
+
+    msq_template = (
+        "{} is not set in the connection context, but required for JWT generation."
+    )
+    if not connection_details.user:
+        raise UsageError(msq_template.format("User"))
+    if not connection_details.account:
+        raise UsageError(msq_template.format("Account"))
+    if not connection_details.private_key_file:
+        raise UsageError(msq_template.format("Private key file"))
```

Comment:
No password?

Reply:
What do you mean?

Reply:
We are not able to provide password? We have to use private key?

Reply:
Yes, only private key is neccessary